aggregator/aggregator_test.go

```diff
@@ -0,0 +1,1576 @@
+
+	// Test data
+	rollbackData := synchronizer.RollbackBatchesData{
+		LastBatchNumber: 100,
+	}
+
+	mockStreamClient.On("IsStarted").Return(true).Once()
+	mockStreamClient.On("ResetProcessEntryFunc").Return().Once()
+	mockStreamClient.On("SetProcessEntryFunc", mock.Anything).Return().Once()
+	mockStreamClient.On("ExecCommandStop").Return(nil).Once()
+	mockStreamClient.On("Start").Return(nil).Once()
+	mockStreamClient.On("ExecCommandStartBookmark", mock.Anything).Return(nil).Once()
+	mockEtherman.On("GetLatestVerifiedBatchNum").Return(uint64(90), nil).Once()
+	mockState.On("DeleteBatchesNewerThanBatchNumber", mock.Anything, rollbackData.LastBatchNumber, nil).Return(nil).Once()
+	mockState.On("DeleteBatchesOlderThanBatchNumber", mock.Anything, rollbackData.LastBatchNumber, nil).Return(nil).Once()
+	mockState.On("DeleteUngeneratedProofs", mock.Anything, nil).Return(nil).Once()
+	mockState.On("DeleteGeneratedProofs", mock.Anything, rollbackData.LastBatchNumber+1, mock.AnythingOfType("uint64"), nil).Return(nil).Once()
+
+	a := Aggregator{
+		ctx:                    context.Background(),
+		streamClient:           mockStreamClient,
+		etherman:               mockEtherman,
+		state:                  mockState,
+		logger:                 log.GetDefaultLogger(),
+		halted:                 atomic.Bool{},
+		streamClientMutex:      &sync.Mutex{},
+		currentBatchStreamData: []byte{},
+		currentStreamBatchRaw:  state.BatchRawV2{},
+		currentStreamL2Block:   state.L2BlockRaw{},
+	}
+
+	a.halted.Store(false)
+	a.handleRollbackBatches(rollbackData)
+
+	assert.False(t, a.halted.Load())
+	mockStreamClient.AssertExpectations(t)
+	mockEtherman.AssertExpectations(t)
+	mockState.AssertExpectations(t)
+}
+
+func Test_handleReceivedDataStream_BATCH_START(t *testing.T) {
+	t.Parallel()
+
+	mockState := new(mocks.StateInterfaceMock)
+	mockL1Syncr := new(mocks.SynchronizerInterfaceMock)
+	agg := Aggregator{
+		state:              mockState,
+		l1Syncr:            mockL1Syncr,
+		logger:             log.GetDefaultLogger(),
+		halted:             atomic.Bool{},
+		currentStreamBatch: state.Batch{},
+	}
+
+	// Prepare a FileEntry for Batch Start
+	batchStartData, err := proto.Marshal(&datastream.BatchStart{
+		Number:  1,
+		ChainId: 2,
+		ForkId:  3,
+		Type:    datastream.BatchType_BATCH_TYPE_REGULAR,
+	})
+	assert.NoError(t, err)
+
+	batchStartEntry := &datastreamer.FileEntry{
+		Type: datastreamer.EntryType(datastream.EntryType_ENTRY_TYPE_BATCH_START),
+		Data: batchStartData,
+	}
+
+	// Test the handleReceivedDataStream for Batch Start
+	err = agg.handleReceivedDataStream(batchStartEntry, nil, nil)
+	assert.NoError(t, err)
+
+	assert.Equal(t, agg.currentStreamBatch.BatchNumber, uint64(1))
+	assert.Equal(t, agg.currentStreamBatch.ChainID, uint64(2))
+	assert.Equal(t, agg.currentStreamBatch.ForkID, uint64(3))
+	assert.Equal(t, agg.currentStreamBatch.Type, datastream.BatchType_BATCH_TYPE_REGULAR)
+}
+
+func Test_handleReceivedDataStreamBatchEnd(t *testing.T) {
```

Comment:
Would it be possible to consolidate all the `handleReceivedDataStream` to a single table unit test? Although I'm aware the setup is pretty different, depending which stream data is received.
```suggestion
func Test_handleReceivedDataStream_BatchEnd(t *testing.T) {
```